### DISCUSSION TITLE
Roller shutter control

### DISCUSSION THREAD

**Discussion by dippes** [arendst/Tasmota#22457] [Support request]
Hello
I have flashed and calibrated a Shelly Plus 2 PM with Tasmota. This is to be used as a shutter control. I have already entered set option 80 1 in the console.

What else needs to be done for the shutter control to work?

**Comment by dippes** [comment]
The roller shutter control works up and down.

How do I set the roller shutter to go down 50%, for example?
How is this controlled? Is there a slider for this in the WIF?

**Reply by Mark-Muc** [reply]
hello . I use this command to set up a shutter (shutternode 1) correctly. 
@ least of a reboot of the device the "shutter slider" shows up and "shutter commands" are working in the console.

Shutter define:
`backlog Interlock 1,2; Interlock ON; SetOption80 1;Shuttermode 1; ShutterRelay1 1`
Hope it helps
Mark_Muc
 PS: other commands I use for set up quickly:
windows time: 
`backlog ShutterOpenDuration1 22; ShutterCloseDuration1 21.5; ShutterSetHalfway1 65; ShutterEnableEndStopTime1`

door time: 
`backlog ShutterOpenDuration1 29; ShutterCloseDuration1 28; ShutterSetHalfway1 59; ShutterEnableEndStopTime1`

**Comment by dippes** [comment]
I defined my Shelly 2 Plus PM like this.

backlog Interlock 1,2; Interlock ON; SetOption80 1;Shuttermode 1; ShutterRelay1 1


Up and down with via the WIF works.

But I can't stop the roller shutter at a certain point via the WIF.this works via the Tasmota Remota app.also manually via the roller shutter switches.what do I have to set so that it also works via the WIF?

**Reply by Mark-Muc** [reply]
@dippes .. my shutters stops if I use any shutter buttons when moving..
to set specific position you can use console for testing .. the command I.e.
shutterposition 50
moves the shutter in the middle position, if the runtimes are set correctly.
you can use a dummy "relais" to get a UI button and in a small rule you can define a shutterposition to move to
Hope it helps
Mark-Muc

**Comment by dippes** [comment]
I'm doing something wrong.....

The shutter needs 20.20 sec to go down and 21 sec to go up.
`backlog ShutterOpenDuration1 21; ShutterCloseDuration2 20.20; ShutterSetHalfway1 65; ShutterEnableEndStopTime1`


But when “shutterposition 50” is entered, the shutter moves down completely. If I then enter “shutterposition 25”, nothing happens.

`10:32:31.074 CMD: shutterposition 50
10:32:31.084 RSL: RESULT = {"POWER2":"ON"}
10:32:31.085 RSL: POWER2 = ON
10:32:31.088 RSL: RESULT = {"Shutter1":{"Position":100,"Direction":-1,"Target":50,"Tilt":0}}
10:32:31.916 RSL: RESULT = {"Shutter1":{"Position":94,"Direction":-1,"Target":50,"Tilt":0}}
10:32:32.888 RSL: RESULT = {"Shutter1":{"Position":88,"Direction":-1,"Target":50,"Tilt":0}}
10:32:33.915 RSL: RESULT = {"Shutter1":{"Position":80,"Direction":-1,"Target":50,"Tilt":0}}
10:32:34.934 RSL: RESULT = {"Shutter1":{"Position":73,"Direction":-1,"Target":50,"Tilt":0}}
10:32:35.910 RSL: RESULT = {"Shutter1":{"Position":66,"Direction":-1,"Target":50,"Tilt":0}}
10:32:36.920 RSL: RESULT = {"Shutter1":{"Position":59,"Direction":-1,"Target":50,"Tilt":0}}
10:32:37.898 RSL: RESULT = {"Shutter1":{"Position":52,"Direction":-1,"Target":50,"Tilt":0}}
10:32:38.196 RSL: RESULT = {"POWER2":"ON"}
10:32:38.198 RSL: POWER2 = ON
10:32:38.200 RSL: SHUTTER1 = 50
10:32:38.202 RSL: RESULT = {"Shutter1":{"Position":50,"Direction":0,"Target":50,"Tilt":0}}
10:33:08.214 RSL: RESULT = {"POWER2":"OFF"}
10:33:08.216 RSL: POWER2 = OFF
10:33:43.279 CMD: shutterposition 50
10:33:43.284 RSL: RESULT = {"Shutter1":{"Position":50,"Direction":0,"Target":50,"Tilt":0}}
10:34:16.248 CMD: shutterposition 25
10:34:16.259 RSL: RESULT = {"POWER2":"ON"}
10:34:16.260 RSL: POWER2 = ON
10:34:16.263 RSL: RESULT = {"Shutter1":{"Position":50,"Direction":-1,"Target":25,"Tilt":0}}
10:34:16.902 RSL: RESULT = {"Shutter1":{"Position":46,"Direction":-1,"Target":25,"Tilt":0}}
10:34:17.915 RSL: RESULT = {"Shutter1":{"Position":39,"Direction":-1,"Target":25,"Tilt":0}}
10:34:18.902 RSL: RESULT = {"Shutter1":{"Position":31,"Direction":-1,"Target":25,"Tilt":0}}
10:34:19.870 RSL: RESULT = {"POWER2":"ON"}
10:34:19.872 RSL: POWER2 = ON
10:34:19.874 RSL: SHUTTER1 = 25
10:34:19.876 RSL: RESULT = {"Shutter1":{"Position":25,"Direction":0,"Target":25,"Tilt":0}}
10:34:31.309 CMD: shutterposition 25
10:34:31.315 RSL: RESULT = {"Shutter1":{"Position":25,"Direction":0,"Target":25,"Tilt":0}}
10:34:49.894 RSL: RESULT = {"POWER2":"OFF"}
10:34:49.896 RSL: POWER2 = OFF`

**Reply by Mark-Muc** [reply]
Hmm I really don't know what's wrong with your device / configuration.
Do you see the shutter slider?
Show you configuration of the GPIO's.
Regards
Mark-Muc

**Reply by dippes** [reply]
Yes, the slider is present.

According to the screenshot, the slider shows that the shutter is approx. 70% open, but in reality it is 100% open.


![GPIO](https://github.com/user-attachments/assets/9a0b23ae-9a0e-469f-97a4-96eae7ac3978)
![Shutterslider](https://github.com/user-attachments/assets/40e320b1-33d1-4ce2-aa74-d6bd7fb38468)

**Reply by Sfre12** [reply]
Two things..
First, your problem sounds, like you did not make a proper calibarion
second, Update to the latest Tasmota version, then you get "better" buttons for the shutter to the UI. (Only the marked area)

![image](https://github.com/user-attachments/assets/ad44fba0-a068-4b38-a735-adb74de672f6)

**Comment by Mark-Muc** [comment]
hmm try to move down the shutter .. close it.. then the following command in the console:
shuttersetclose
shuttersetup
the shutter then should open and close again and the shutter times should set correctly ..
Mark-Muc

**Reply by Sfre12** [reply]
I am sorry, i do not have any shutter with power meter...So I never used shutter setup. I do it manually every time
Check with command shuttercloseduration and shutteropenduration what is the setting, and verify this time!

**Comment by dippes** [comment]
I entered shutterclose.
shuttersetup opened the shutter but did not close it again.

```
16:38:45.529 CMD: shuttersetup
16:38:45.533 SHT: ShutterMode: 1
16:38:45.535 SHT: Setup: Ensure shutter is close. Now open, autostop detect. max duration is 2min Phase:2
16:38:46.059 MTR: =Saved     1 fabric(s) and 5 session(s)
16:38:46.549 MTR: New Connection (CASE id=33843) from [192.168.1.119]:5541
16:38:46.596 RSL: RESULT = {"POWER1":"ON"}
16:38:46.598 RSL: POWER1 = ON
16:38:46.600 RSL: RESULT = {"Shutter1":{"Position":0,"Direction":1,"Target":100,"Tilt":0}}
16:38:47.245 RSL: RESULT = {"Shutter1":{"Position":0,"Direction":1,"Target":100,"Tilt":0}}
16:38:47.264 SHT: Setup. Start detected. Waiting for STOP
16:38:47.998 MTR: =Saved     1 fabric(s) and 5 session(s)
16:38:48.042 RSL: RESULT = {"Shutter1":{"Position":0,"Direction":1,"Target":100,"Tilt":0}}
16:38:49.451 RSL: RESULT = {"Shutter1":{"Position":0,"Direction":1,"Target":100,"Tilt":0}}
16:38:50.341 QPC: Reset
16:38:50.351 RSL: STATE = {"Time":"2024-11-26T16:38:50","Uptime":"0T00:00:09","UptimeSec":9,"Heap":86,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":382,"MqttCount":0,"Berry":{"HeapUsed":45,"Objects":553},"POWER1":"ON","POWER2":"OFF","Wifi":{"AP":1,"SSId":"Router966F24E92D65857CBC","BSSId":"48:5D:35:A5:4A:C3","Channel":13,"Mode":"HT40","RSSI":42,"Signal":-79,"LinkCount":1,"Downtime":"0T00:00:03"}}
16:38:50.401 RSL: SENSOR = {"Time":"2024-11-26T16:38:50","Switch1":"ON","Switch2":"OFF","ANALOG":{"Temperature1":9.7},"ENERGY":{"TotalStartTime":"2024-11-10T12:28:20","Total":0.045,"Yesterday":0.004,"Today":0.006,"Period":[0,0],"Power":[0,0],"ApparentPower":[0,0],"ReactivePower":[0,0],"Factor":[0.00,0.00],"Frequency":50,"Voltage":228,"Current":[0.000,0.000]},"Shutter1":{"Position":0,"Direction":1,"Target":100,"Tilt":0},"TempUnit":"C"}
16:38:50.447 RSL: RESULT = {"Shutter1":{"Position":0,"Direction":1,"Target":100,"Tilt":0}}
16:38:50.935 RSL: RESULT = {"Shutter1":{"Position":0,"Direction":1,"Target":100,"Tilt":0}}
16:38:51.815 SHT: 105 stopp_armed:1
16:38:52.303 SHT: 114 stopp_armed:2
16:38:52.321 RSL: RESULT = {"Shutter1":{"Position":1,"Direction":1,"Target":100,"Tilt":0}}
16:38:52.732 SHT: 123 stopp_armed:3
16:38:52.786 SHT: 124 stopp_armed:4
16:38:52.833 SHT: 125 stopp_armed:5
16:38:53.591 RSL: RESULT = {"Shutter1":{"Position":1,"Direction":1,"Target":100,"Tilt":0}}
16:38:53.803 SHT: 144 stopp_armed:6
16:38:53.807 RSL: RESULT = {"POWER1":"ON"}
16:38:53.809 RSL: POWER1 = ON
16:38:53.811 RSL: SHUTTER1 = 1
16:38:53.814 RSL: RESULT = {"Shutter1":{"Position":1,"Direction":0,"Target":1,"Tilt":0}}
16:38:53.832 SHT: Setup. Cycletime is: 5.2 sec
16:38:53.833 SHT: ShutterMode: 1
16:38:53.835 SHT: Setup: Now close, autostop detect. max duration is 2min
16:39:23.914 RSL: RESULT = {"POWER1":"OFF"}
16:39:23.916 RSL: POWER1 = OFF
16:39:48.222 MTR: =Saved     1 fabric(s) and 5 session(s)
16:40:54.372 RSL: RESULT = {"Shutter1":{"Position":100,"Direction":0,"Target":100,"Tilt":0}}`
```

**Comment by dippes** [comment]
How can I view and delete the shutter commands?

**Comment by dippes** [comment]
I found this error message in the console, where did it come from?

`11:41:55.406 MQT: stat/tasmota_8815D8/RESULT = {"Command":"Unknown","Input":"PULSETIMESHUTTER"}`

**Reply by sfromis** [reply]
It comes from you trying to execute such an invalid command name. With `Weblog 3` enabled, you'll also get a log line with `SRC:` indicating how you issued the command.
